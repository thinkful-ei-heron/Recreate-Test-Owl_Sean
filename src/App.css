main{
    padding:0;
    display:flex;
    height:100%;
}
