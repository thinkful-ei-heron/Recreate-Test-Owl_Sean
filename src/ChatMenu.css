.menu {
    border-style:solid;
    border-width:5px;
    width:20.7vw;
    height:9%;
    padding:0;
    text-align: center;
    position:fixed;
    bottom:91%;
    left:0;
    display:flex;
    justify-content: space-around;
}

button {
    display:flex;
    margin-top:5%;
    height:50%;
    flex-wrap: nowrap;
    overflow:hidden;
    font-size:0.35em;
    padding:0;
}

form{
    position:fixed;
    bottom:10.5%;
    margin-left:1%;
}