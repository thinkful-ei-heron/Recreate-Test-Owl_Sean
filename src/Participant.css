*{
    box-sizing: border-box;
}

.outerParticipantContainer{
    display:flex;
    position: fixed;
    flex-direction: column;
    margin:0;
    height:81.5%;
    border-style: solid;
    border-width: 5px;
    border-color: red;
    width:20.7vw;
    justify-content: space-evenly;
    bottom:9.5%; 
    padding: 1%;
}

.chatContainer{
    display:flex;
    position: fixed;
    flex-direction: column;
    margin:0;
    height:81.5%;
    border-style: solid;
    border-width: 5px;
    border-color: red;
    width:20.7vw;
    justify-content: space-evenly;
    bottom:9.5%; 
    padding: 1%;
}

.chatE{
    font-size: 0.5em;
}

ul{
    margin:0;
}

.participantBox
::-webkit-scrollbar {
    width: 10px;
  }

  ::-webkit-scrollbar-track {
    background: #f1f1f1; 
  }

  ::-webkit-scrollbar-thumb {
    background: #888; 
  }

  h2 {
      display:block;
      margin:0;
      font-size: 1.0em;
  }

  img {
      max-height:4vw;
      max-width:4vw;
      background-color: aqua;
      border-radius: 50px;
      /* left:5vw; */
  }

  .participant{
      width: 100%;
      height:80%;
      display: flex;
      justify-content: space-around;
      flex-direction: column;
  }

  .avatarSeperator{
      width:100%;
  }

  .participantBox {
      display:flex;
      width:100%;
      text-align: center;
      margin-bottom:1vw;
      margin:0;
      justify-content: space-between;
  }

  p{
      margin:0;
      font-size:0.9em;
  }

  .participant > button{
      margin-top:0;
      margin-bottom:0;
  }

  .chatInput{
      margin-left:auto;
      margin-right:auto;
      width:17vw;
      font-size: 2em;
  }

  form{
      width:20.7vw;
  }

  textarea{
    width:17vw;
    font-size: 2em;
    height:100%;
  }

