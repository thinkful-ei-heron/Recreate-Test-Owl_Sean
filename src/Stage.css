.stage{
    position:fixed;
    background-color: #404853;
    bottom:51vw;
    left:20.7vw;
    display:flex;
    flex-wrap: wrap;
    height:90%;
    flex-direction:column;
    justify-content: space-around;
    width:80vw;
    top:0;
}

.stageBox{
    border-style:solid;
    border-color:red;
    height:25%;
    display:block;
    width:25%;
    margin-left:auto;
    margin-right:auto;
    text-align: center;
}

img{
    display:block;
}
