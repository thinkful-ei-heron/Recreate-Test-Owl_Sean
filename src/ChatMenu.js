import React from 'react';
import './ChatMenu.css';

class ChatMenu extends React.Component {

    handleChatEvent=()=>{
        //console.log('hello')
        this.props.handleChatEvent()
    }

    handleParticipantEvent=()=>{
        //console.log('there')
        this.props.handleParticipantEvent()
    }
    
    
    render(props) {
        
        return (
            <div className="menu">
                <button type="button" onClick={() => this.handleChatEvent()}>Chat</button>
                <button type="button" onClick={() =>this.handleParticipantEvent()}>Participants</button>
                <button type="button">Settings</button>
                <button type="button">Copy Session Url</button>
                <button type="button">See FAQ</button>
            </div>
        );
    }
}

export default ChatMenu