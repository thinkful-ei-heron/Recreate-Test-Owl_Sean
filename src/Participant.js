import React from 'react';
import './Participant.css';
import chatEvents from './chatEvents'
import participantList from './participantList'

class Participant extends React.Component {
    
    constructor(props){
        super(props);
        
        this.state={
            page:'participantScreen',
            chat:chatEvents,
            input:''
        }
   
    }
    static defaultProps = {
        screen:'participantScreen',
        list: participantList
    }

    componentDidMount(){
        this.setState({page: this.props.screen})
    }

    componentWillReceiveProps=()=>{
        this.setState({page: this.props.screen})
    }

    handleText=(event)=>{
        this.setState({
            input:event.target.value
        })
    }
    
    screenChoice=()=>{
            let current = participantList.list ;
            let filtered = current.filter(participant => participant.inSession === true);
            let display = filtered.map((filtered) => 
            <div className="participantBox" key={filtered.id}>
                <img src={filtered.avatar} alt='avatar' />
                <div className="avatarSeperator">
                    <h2 key={filtered.id}>{filtered.name}</h2>
                    <p>{filtered.onStage === true? "onStage" : "inSession" }</p>
                </div>
            </div>
        );
      

        if(this.state.page == 'participantScreen' || this.state.page == ''){
            return(
             <div className="outerParticipantContainer">
                <div className="participant" key={filtered.id}>
                    {display}  
                </div>  
             </div>  
            )
        } else if(this.state.page == 'chatScreen'){
            return(
                
                <div className='chatContainer'>
                    <form>
                        {this.state.chat ? this.state.chat.chatEvents.map((chat, index) => {
                            return(
                                <ul key={index}>
                                    {chat.message}
                                    {chat.type === 'thumbs-up' ? <span className='chatE'>'thumbs-up'</span> : null}
                                    {chat.type === 'thumbs-down' ? <span className='chatE'>'thumbs-down' </span>: null}
                                    {chat.type === 'raise-hand' ? <span className='chatE'>'A hand is raised'</span> : null}
                                    {chat.type === 'clap' ? <span className='chatE'>'Clap!' </span>: null}
                                    {chat.type === 'join' ? <span className='chatE'>'join'</span> : null}
                                    {chat.type === 'leave' ? <span className='chatE'>'leave'</span> : null}
                                    {chat.type === 'join-stage' ? <span className='chatE'>'join-stage'</span> : null}
                                    {chat.type === 'leave-stage' ? <span className='chatE'>'leave-stage' </span>: null}
                                </ul>
                            )
                        }) : null}
                        <div className='inputText'>{this.state.input}</div>
                  
        <input type="text" className="chatInput" name="chat" onChange={this.handleText}></input>
                    </form>
                </div>
            )
        }
    }
    
    render(){
        


        return(
            <div>{this.props.list ? this.screenChoice() : null}</div>   
        )
    }
}

export default Participant