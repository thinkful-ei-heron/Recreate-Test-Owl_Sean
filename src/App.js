import React from 'react';
import './App.css'
import Participant from './Participant';
import participantList from './participantList';
import Stage from './Stage';
import StageBottom from './StageBottom';
import chatEvents from './chatEvents';
import ChatMenu from './ChatMenu';


class App extends React.Component {

  constructor(props) {
    super(props)
      this.state = {
        participants:{participantList},
        chat:{chatEvents},
        current:'',
  }
}

  handleChatEvent=(event)=>{
     this.setState({
       current : 'chatScreen'
      })
     console.log(this.state.current)
  }

  handleParticipantEvent=(event)=>{
    this.setState({
      current : 'participantScreen'
     })
    console.log(this.state.current)
 }


render(){
  return (
    <main className="App" >
      <Participant list={this.state.participants} screen={this.state.current}/>
      <Stage list={this.state.participants}/>
      <StageBottom />
      <ChatMenu handleChatEvent={this.handleChatEvent} handleParticipantEvent={this.handleParticipantEvent} />
    </main>
  );
}
}

export default App;
