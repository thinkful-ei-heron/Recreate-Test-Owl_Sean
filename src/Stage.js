import React from 'react';
import './Stage.css';
import participantList from './participantList'



class Stage extends React.Component {

    static defaultProps = {
        list: participantList
    }
     
    render(props) {
        let current = participantList.list;
        let filtered = current.filter(participant => participant.onStage === true);

        let display = filtered.map((filtered) => 
            <div className="stageBox" key={filtered.id}>
                <h2 key={filtered.id}>{filtered.name}</h2>
                <img src={filtered.avatar} alt='avatar' />
            </div>
        );

        return(
            <div className="stage">
                {display}
            </div>
        );
     }
}

export default Stage