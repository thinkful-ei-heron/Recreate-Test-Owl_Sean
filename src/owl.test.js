/* eslint-disable no-unused-expressions */
import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import participantList from './participantList'
import App from './App'
import ChatMenu from './ChatMenu'
import Participant from './Participant'
import Stage from './Stage'
import StageBottom from './StageBottom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'



describe(`App Component`, () => {
    it(`Renders without errors`, () => {
        const div = document.createElement('div');
        ReactDOM.render(<App /> , div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it(`Shows the App component (a container) when rendered`, () => {
        const tree = renderer
            .create(<App />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

});

describe(`ChatMenu Component`, () => {
    it(`Renders without errors`, () => {
        const div = document.createElement('div');
        ReactDOM.render(<ChatMenu /> , div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it(`Shows the ChatMenu component when rendered`, () => {
        const tree = renderer
            .create(<ChatMenu />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})

describe(`Participant Component`,  () => {
     it(`Renders without errors`, () => {
        const div = document.createElement('div');
        ReactDOM.render(<Participant list={participantList}/> , div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it(`Shows the Participant component when rendered`, () => {
        const tree = renderer
            .create(<Participant list={participantList}/>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})

describe(`Stage Component`, () => {
    it(`Renders without errors`, () => {
        const div = document.createElement('div');
        ReactDOM.render(<Stage list={participantList}/> , div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it(`Shows the Stage component when rendered`, () => {
        const wrapper = shallow(<Stage list={participantList}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})

describe(`StageBottom Component`, () => {
    it(`Renders without errors`, () => {
        const div = document.createElement('div');
        ReactDOM.render(<StageBottom /> , div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it(`Shows the StageBottom component when rendered`, () => {
        const tree = renderer
            .create(<StageBottom />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})