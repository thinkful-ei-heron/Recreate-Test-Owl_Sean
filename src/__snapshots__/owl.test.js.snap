// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Shows the App component (a container) when rendered 1`] = `
<main
  className="App"
>
  <div>
    <div
      className="outerParticipantContainer"
    >
      <div
        className="participant"
      >
        <div
          className="participantBox"
        >
          <img
            alt="avatar"
            src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
          />
          <div
            className="avatarSeperator"
          >
            <h2>
              Koren Templeton
            </h2>
            <p>
              onStage
            </p>
          </div>
        </div>
        <div
          className="participantBox"
        >
          <img
            alt="avatar"
            src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
          />
          <div
            className="avatarSeperator"
          >
            <h2>
              Caty Flucker
            </h2>
            <p>
              inSession
            </p>
          </div>
        </div>
        <div
          className="participantBox"
        >
          <img
            alt="avatar"
            src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
          />
          <div
            className="avatarSeperator"
          >
            <h2>
              Frank Runciman
            </h2>
            <p>
              inSession
            </p>
          </div>
        </div>
        <div
          className="participantBox"
        >
          <img
            alt="avatar"
            src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
          />
          <div
            className="avatarSeperator"
          >
            <h2>
              Ashla Attwool
            </h2>
            <p>
              onStage
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="stage"
  >
    <div
      className="stageBox"
    >
      <h2>
        Koren Templeton
      </h2>
      <img
        alt="avatar"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="stageBox"
    >
      <h2>
        Ashla Attwool
      </h2>
      <img
        alt="avatar"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
    </div>
  </div>
  <div
    className="stageBottom"
  >
    <button
      className="leaveStageButton"
      type="button"
    >
      Leave Stage
    </button>
  </div>
  <div
    className="menu"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Chat
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Participants
    </button>
    <button
      type="button"
    >
      Settings
    </button>
    <button
      type="button"
    >
      Copy Session Url
    </button>
    <button
      type="button"
    >
      See FAQ
    </button>
  </div>
</main>
`;

exports[`ChatMenu Component Shows the ChatMenu component when rendered 1`] = `
<div
  className="menu"
>
  <button
    onClick={[Function]}
    type="button"
  >
    Chat
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Participants
  </button>
  <button
    type="button"
  >
    Settings
  </button>
  <button
    type="button"
  >
    Copy Session Url
  </button>
  <button
    type="button"
  >
    See FAQ
  </button>
</div>
`;

exports[`Participant Component Shows the Participant component when rendered 1`] = `
<div>
  <div
    className="outerParticipantContainer"
  >
    <div
      className="participant"
    >
      <div
        className="participantBox"
      >
        <img
          alt="avatar"
          src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
        />
        <div
          className="avatarSeperator"
        >
          <h2>
            Koren Templeton
          </h2>
          <p>
            onStage
          </p>
        </div>
      </div>
      <div
        className="participantBox"
      >
        <img
          alt="avatar"
          src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
        />
        <div
          className="avatarSeperator"
        >
          <h2>
            Caty Flucker
          </h2>
          <p>
            inSession
          </p>
        </div>
      </div>
      <div
        className="participantBox"
      >
        <img
          alt="avatar"
          src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
        />
        <div
          className="avatarSeperator"
        >
          <h2>
            Frank Runciman
          </h2>
          <p>
            inSession
          </p>
        </div>
      </div>
      <div
        className="participantBox"
      >
        <img
          alt="avatar"
          src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
        />
        <div
          className="avatarSeperator"
        >
          <h2>
            Ashla Attwool
          </h2>
          <p>
            onStage
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Stage Component Shows the Stage component when rendered 1`] = `
<div
  className="stage"
>
  <div
    className="stageBox"
    key="1"
  >
    <h2
      key="1"
    >
      Koren Templeton
    </h2>
    <img
      alt="avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="stageBox"
    key="5"
  >
    <h2
      key="5"
    >
      Ashla Attwool
    </h2>
    <img
      alt="avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
  </div>
</div>
`;

exports[`StageBottom Component Shows the StageBottom component when rendered 1`] = `
<div
  className="stageBottom"
>
  <button
    className="leaveStageButton"
    type="button"
  >
    Leave Stage
  </button>
</div>
`;
